futures-util/src/try_stream/try_next.rs

```diff
@@ -0,0 +1,38 @@
+use futures_core::future::Future;
+use futures_core::stream::TryStream;
+use futures_core::task::{self, Poll};
+use core::marker::Unpin;
+use core::mem::PinMut;
+
+/// A future which attempts to collect all of the values of a stream.
+///
+/// This future is created by the `Stream::try_collect` method.
+#[derive(Debug)]
+#[must_use = "streams do nothing unless polled"]
+pub struct TryNext<'a, St: 'a> where St: TryStream {
+    stream: &'a mut St,
+}
+
+impl<'a, St: TryStream + Unpin> Unpin for TryNext<'a, St> {}
+
+impl<'b, St: TryStream + Unpin> TryNext<'b, St> {
```

Comment:
Why is the lifetime called `'b` here?

Reply:
The `unsafe_pinned` macro introduces a lifetime called `'a` which would overlap with this one. We should probably change that to introduce some other less-likely-to-conflict lifetime name.

Reply:
Ah I see